app/dashboard/src/components/dashboard/AssetRow.tsx

```diff
@@ -160,12 +161,16 @@ export const AssetRow = React.memo(function AssetRow(props: AssetRowProps) {
   const grabKeyboardFocusRef = useSyncRef(grabKeyboardFocus)
   const asset = item.item
   const [insertionVisibility, setInsertionVisibility] = React.useState(Visibility.visible)
-  const [rowState, setRowState] = React.useState<assetsTable.AssetRowState>(() =>
-    object.merge(assetRowUtils.INITIAL_ROW_STATE, {
-      setVisibility: setInsertionVisibility,
-      isEditingName: driveStore.getState().newestFolderId === asset.id,
-    }),
+  const [innerRowState, setRowState] = React.useState<assetsTable.AssetRowState>(() =>
+    object.merge(assetRowUtils.INITIAL_ROW_STATE, { setVisibility: setInsertionVisibility }),
   )
+
+  const isEditingName = useNewestFolderId() === asset.id
```

Comment:
it's intentional that we do not use `useNewestFolderId` directly, since it causes way too many unnecessary updates. `useStore` with a boolean selector should be fine though.